drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/DMNTypeServiceImpl.java

```diff
@@ -85,103 +81,190 @@ public DMNRuntime getDMNRuntime(Path path) {
-        genericTypeInfoMap.put(name, Collections.singletonList(genericKey));
-        simpleFields.put(name, List.class.getCanonicalName());
-
-        return genericKey;
+    /**
+     * Creates a <code>FactModelTree</code> for <code>DMNType</code> where <code>DMNType.isComposite()</code> == <code>true</code>
+     * @param name
+     * @param fullPropertyPath
+     * @param type
+     * @param hiddenFacts
+     * @param fmType
+     * @throws WrongDMNTypeException if <code>DMNType.isComposite()</code> != <code>true</code>
+     */
+    protected FactModelTree createFactModelTreeForComposite(Map<String, List<String>> genericTypeInfoMap, String name, String fullPropertyPath, DMNType type, SortedMap<String, FactModelTree> hiddenFacts, FactModelTree.Type fmType) throws WrongDMNTypeException {
+        if (!type.isComposite()) {
+            throw new WrongDMNTypeException();
+        }
+        Map<String, String> simpleFields = new HashMap<>();
+        FactModelTree toReturn = new FactModelTree(name, "", simpleFields, genericTypeInfoMap, fmType);
+        for (Map.Entry<String, DMNType> entry : type.getFields().entrySet()) {
+            String expandablePropertyName = fullPropertyPath + "." + entry.getKey();
+            if (entry.getValue().isCollection()) {  // if it is a collection, generate the generic and add as hidden fact a simple or composite fact model tree
+                FactModelTree fact = createInnerFactModelTree(new HashMap<>(), entry.getKey(), expandablePropertyName, entry.getValue(), hiddenFacts, FactModelTree.Type.UNDEFINED);
+                hiddenFacts.put(entry.getKey(), fact);
+                simpleFields.put(entry.getKey(), List.class.getCanonicalName());
+                genericTypeInfoMap.put(entry.getKey(), fact.getGenericTypeInfo("value"));
+            } else {
+                if (entry.getValue().isComposite()) { // a complex type needs the expandable property and then in the hidden map, its fact model tree
+                    FactModelTree fact = createInnerFactModelTree(genericTypeInfoMap, entry.getKey(), expandablePropertyName, entry.getValue(), hiddenFacts, FactModelTree.Type.UNDEFINED);
+                    hiddenFacts.put(entry.getValue().getName(), fact);
+                    toReturn.addExpandableProperty(entry.getKey(), entry.getValue().getName());
+                } else {  // a simple type is just name -> type
+                    simpleFields.put(entry.getKey(), entry.getValue().getName());
+                }
+            }
+        }
+        return toReturn;
     }
 
-    protected void checkTypeSupport(DMNType type, ErrorHolder errorHolder, String path) {
+    /**
+     * Check the given <code>DMNType</code> to eventually detect and add errors to given <code>ErrorHolder</code>
+     * @param type
+     * @param errorHolder
+     * @param fullPropertyPath
+     */
+    protected void checkType(DMNType type, ErrorHolder errorHolder, String fullPropertyPath) {
```

Comment:
Personally I prefer `checkTypeSupport` because this method verify if there is any field/object with a not supported structure. I think `checkType` is a bit too generic

Reply:
@danielezonca 
It _was_ checkTypeSupport due to the lack of support for collection, etc. But on one side we are adding this support, on the other side I was thinking to keep it and use as a more generic check/validation.

Reply:
@danielezonca 
_FIXED_